sys/include/fido2/ctap.h

```diff
@@ -0,0 +1,680 @@
+/** @} */
+
+/**
+ * @brief CTAP thread stack size
+ */
+#ifdef CONFIG_FIDO2_CTAP_STACK_SIZE
+#define CTAP_STACKSIZE CONFIG_FIDO2_CTAP_STACK_SIZE
+#else
+#define CTAP_STACKSIZE 15000
+#endif
+
+/**
+ * @brief Max size of relying party name
+ */
+#define CTAP_RP_MAX_NAME_SIZE   32
+
+/**
+ * @brief Max size of username including null character
+ */
+#define CTAP_USER_MAX_NAME_SIZE 64 + 1
+
+/**
+ * @brief Max size of user id
+ */
+#define CTAP_USER_ID_MAX_SIZE 64
+
+/**
+ * @brief Max size of a domain name including null character
+ */
+#define CTAP_DOMAIN_NAME_MAX_SIZE 253 + 1
+
+/**
+ * @brief Max size of icon including null character
+ */
+#define CTAP_ICON_MAX_SIZE  128 + 1
+
+/**
+ * @brief PIN min size
+ */
+#define CTAP_PIN_MIN_SIZE   4
+
+/**
+ * @brief Encrypted newPin min size
+ *
+ * Encrypted PIN is padded with trailing 0x00 bytes to a minimum length
+ * of 64 in order to prevent leak of PIN length.
+ */
+#define CTAP_PIN_ENC_MIN_SIZE 64
+
+/**
+ * @brief Encrypted newPin max size
+ *
+ */
+#define CTAP_PIN_ENC_MAX_SIZE 256
+
+/**
+ * @brief PIN max size
+ */
+#define CTAP_PIN_MAX_SIZE   64
+
+/**
+ * @brief Max total consecutive incorrect PIN attempts
+ */
+#define CTAP_PIN_MAX_ATTS 8
+
+/**
+ * @brief Max consecutive incorrect PIN attempts for 1 boot cycle
+ */
+#define CTAP_PIN_MAX_ATTS_BOOT 3
+
+/**
+ * @brief PIN protocol version
+ */
+#define CTAP_PIN_PROT_VER 1
+
+/**
+ * @brief Total number of supported PIN protocol versions
+ */
+#define CTAP_AMT_SUP_PIN_VER 1
+
+/**
+ * @brief Size of pin token
+ *
+ * Needs to be a multiple of 16 bytes (AES block length)
+ */
+#define CTAP_PIN_TOKEN_SZ 16
+
+/**
+ * @brief Size of pin auth
+ */
+#define CTAP_PIN_AUTH_SZ 16
+
+/**
+ * @brief Size of key used to encrypt credential
+ *
+ * Needed if authenticator is unable to store resident keys.
+ * See webauthn specification (version 20190304) section 4 (Credential ID)
+ * for details
+ */
+#define CTAP_CRED_KEY_LEN 16
+
+/**
+ * @brief AES_CCM_L parameter
+ *
+ * L has to be between 2 and 8. Value of 2 means that message has to be
+ * in the range 0 <= l(m) < 2^(16) = 65536.
+ * This should always be sufficient to send an encrypted resident key.
+ */
+#define CTAP_AES_CCM_L 2
+
+/**
+ * @brief AES CCM nonce size
+ */
+#define CTAP_AES_CCM_NONCE_SIZE (15 - CTAP_AES_CCM_L)
+
+/**
+ * @brief Total size of AES CCM credential id
+ *
+ * Size of encrypted resident key = resident key - cred id
+ *                                        - has_nonce
+ */
+#define CTAP_CREDENTIAL_ID_ENC_SIZE (sizeof(struct ctap_resident_key) - \
+                                     sizeof(((struct ctap_resident_key *)0)-> \
+                                            cred_desc.cred_id) - \
+                                     sizeof(((struct ctap_resident_key *)0)-> \
+                                            cred_desc.has_nonce))
+/**
+ * @brief Start page for storing resident keys
+ */
+#define CTAP_RK_START_PAGE 34U
```

Comment:
I think that was kinda arbitrary. Adjusted it to be 15 for now. 13 is the lowest it can be, before it overwrites parts of the application and causes crashes.

Reply:
I see, so 15 is also kind of "arbitrary". I would leave it like this for now, but add a @note explaining the situation. Furthermore, the fido2 doc.txt should clearly state that keys are currently stored "somewhere" in plain flash.